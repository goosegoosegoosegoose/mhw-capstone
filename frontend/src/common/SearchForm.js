import React, { useState } from "react";
import MhwApi from "../api";
import { Button, Row } from "react-bootstrap"

const SearchForm = ({ type, header, get }) => {
  const [formData, setFormData] = useState("")

  const search = async (handle) => {
    const res = await MhwApi.search(type, header, handle);
    get(res);
  }

  const handleChange = (evt) => {
    const { value } = evt.target;
    setFormData(value);
  };

  const handleSubmit = (evt) => {
    evt.preventDefault();
    search(formData);
    setFormData("");
  };

  return (
    <form onSubmit={handleSubmit}>
      <Row className="my-3">
        <div className="input-group col-2">
          <input
            className="form-control"
            value={formData}
            placeholder="Enter search"
            onChange={handleChange}
          />
          <Button className="mx-1" type="submit" variant="primary" size="sm">Submit</Button>
        </div>
      </Row>
    </form>
  )
}

export default SearchForm;