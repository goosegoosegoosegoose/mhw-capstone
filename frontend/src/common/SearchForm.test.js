import { render } from '@testing-library/react';
import SearchForm from "./SearchForm";

test('smoke test', () => {
  render(      
    <SearchForm />
  );
});

test('snapshot', () => {
  const { asFragment } = render(<SearchForm />);
  expect(asFragment()).toMatchSnapshot();
});