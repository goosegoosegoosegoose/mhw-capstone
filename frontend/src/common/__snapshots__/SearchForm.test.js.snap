// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <form>
    <div
      class="my-3 row"
    >
      <div
        class="input-group col-2"
      >
        <input
          class="form-control"
          placeholder="Enter search"
          value=""
        />
        <button
          class="mx-1 btn btn-dark btn-sm"
          type="submit"
        >
          Submit
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
